lib/matplotlib/colors.py

```diff
@@ -955,6 +955,172 @@ def scaled(self):
         return self.vmin is not None and self.vmax is not None
 
 
+class PiecewiseLinearNorm(Normalize):
+    """
+    Normalizes data into the ``[0.0, 1.0]`` interval over linear segments.
+    """
+    def __init__(self, data_points=None, norm_points=None):
+        """Normalize data linearly between the defined stop points.
+
+        Parameters
+        ----------
+        data_points : tuple of floats
+            Floats spanning the data to be mapped between 0-1
+
+        norm_points : tuple of floats or None (default)
+            Floats spanning [0, 1] that the data points will map to.  If
+            *None*, the data points will be mapped equally into [0, 1].
+
+        Examples
+        --------
+        Note this example is equivalent to the `.DivergingNorm` example.
+        >>> import matplotlib.colors as mcolors
+        >>> offset = mcolors.PiecewiseLinearNorm([-2, 0, 4])
+        >>> data = [-2., -1., 0., 1., 2., 3., 4.]
+        >>> offset(data)
+        array([0., 0.25, 0.5, 0.625, 0.75, 0.875, 1.0])
+        """
+        self._data_points = (np.asarray(data_points)
+                                if data_points is not None
+                                else None)
+        self._norm_points = (np.asarray(norm_points)
+                                if norm_points is not None
+                                else None)
+
+        if not np.all(np.diff(self._data_points) > 0):
```

Comment:
This will fail if None is passed in, even though that case seems explicitly catered for just above.

Reply:
Good point.  Thats really an error - if None is passed, there is no way to change it later, except to change vmin and vmax, in which case the user should just use `Normalize`.  So don't allow `data_points=None` to be a valid  input...

Reply:
OK, I put a trap in.  But I wonder if you are mixing up `_data_points` and `_norm_points`?